galley/pkg/config/schema/codegen/tools/collections.main.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2019 Istio Authors
```

Comment:
Why the weird file name?

Reply:
This is a "main" file for a tool. Notice the build annotation in the file. It doesn't get into compilation with the rest of the code, but gets called via "go:generate go run ..."
Here is an example: https://github.com/ozevren/istio/blob/nwo-2-single/galley/pkg/config/processor/metadata/gen.go